src/ast/passes/semantic_analyser.cpp

```diff
@@ -2672,6 +2672,222 @@ void SemanticAnalyser::visit(AssignVarStatement &assignment)
   }
 }
 
+void SemanticAnalyser::set_uint64_env_var(AssignConfigVarStatement &assignment,
+                                          uint64_t &dest)
+{
+  auto &assignTy = assignment.expr->type;
+  if (!assignTy.IsIntegerTy())
+  {
+    LOG(ERROR, assignment.loc, err_)
+        << "Invalid type for " << assignment.config_var
+        << ". Type: " << assignTy.type;
+  }
+  else
+  {
+    dest = dynamic_cast<Integer *>(assignment.expr)->n;
+  }
+}
+
+void SemanticAnalyser::set_bool_env_var(AssignConfigVarStatement &assignment,
+                                        bool &dest)
+{
+  auto &assignTy = assignment.expr->type;
+  if (!assignTy.IsIntegerTy())
+  {
+    LOG(ERROR, assignment.loc, err_)
+        << "Invalid type for " << assignment.config_var
+        << ". Type: " << assignTy.type;
+  }
+  else
+  {
+    auto val = dynamic_cast<Integer *>(assignment.expr)->n;
+    if (val == 0)
+    {
+      dest = false;
+    }
+    else if (val == 1)
+    {
+      dest = true;
+    }
+    else
+    {
+      LOG(ERROR) << "Invalid value for " << assignment.config_var
+                 << ". Needs to be 0 or 1. Value: " << val;
+    }
+  }
+}
+
+std::string SemanticAnalyser::get_env_ident(const std::string &raw_ident)
+{
+  std::string env_ident = raw_ident;
+  static const std::string prefix = "BPFTRACE_";
+  std::transform(env_ident.begin(),
+                 env_ident.end(),
+                 env_ident.begin(),
+                 [](unsigned char c) { return std::toupper(c); });
+
+  if (!env_ident.rfind(prefix, 0) == 0)
+  {
+    env_ident.insert(0, prefix);
+  }
+  return env_ident;
+}
+
+void SemanticAnalyser::visit(AssignConfigVarStatement &assignment)
+{
+  assignment.expr->accept(*this);
+
+  std::string raw_ident = assignment.config_var;
+  std::string env_ident = get_env_ident(raw_ident);
+  bool env_is_set = false;
+  auto &assignTy = assignment.expr->type;
+
+  auto invalidType = [&]() {
+    LOG(ERROR, assignment.loc, err_)
+        << "Invalid type for " << raw_ident << ". Type: " << assignTy.type;
+  };
+
+  auto matches = [&](std::string &&env_var) -> bool {
+    if (env_ident == env_var)
+    {
+      env_is_set = std::getenv(env_var.c_str()) != nullptr;
+      return true;
+    }
+    return false;
+  };
+
+  if (matches("BPFTRACE_STACK_MODE"))
```

Comment:
This is probably super-nit but since we want to encourage moving from env vars to config options, we should not use the env var name rather than the config name (i.e. `"stack_mode"`) here. Also, if we introduce new config options without corresponding env vars inn future, the code will remain consistent.